src/lib/schematics/update/update-schematic.md

```diff
@@ -0,0 +1,169 @@
+# Update schematic
+
+The update schematic consists of multiple migration entry-points whereas every entry-point targets a specific Angular Material version.
+
+As of right now, we have two migration entry-points that handle the breaking changes for the given target version:
+
+| Target Version | Description |
+|----------------|-------------|
+| V6 | Upgrade from any version to Angular Material 6.0.0 |
+| V7 | Upgrade from any version to Angular Material 7.0.0 |
+
+Note that the migrations run in order if both versions transitively targeted. For example, consider an application which uses Angular Material 5.0.0. In case the developer runs `ng update`, the  Angular CLI installs V6 and applies the migrations. Then after completion, it will install V7 and apply the associated V7 migrations.
+
+## Update concept
+
+The goal of the update schematic is to automatically migrate code that is affected by breaking changes of the target version. Most of the time, we can apply such automatic migrations, but there are also a few breaking changes that cannot be migrated automatically. 
+
+In that case, our goal should be to notify the developer about the breaking change that needs developer attention.
+
+## Source file transformation
+
+There are various approaches to applying changes to developer's source files:
```

Comment:
I would reorganize this section a bit, structured like

* We use TSLint because have has these properties: ...
* We don't use these other approaches because: ...